rust/src/lib.rs

```diff
@@ -508,17 +508,22 @@ pub fn clear_cache(log: &Logger) {
     }
 }
 
-pub fn create_default_http_client() -> Client {
-    http_client_builder()
-        .timeout(Duration::from_secs(REQUEST_TIMEOUT_SEC))
-        .build()
-        .unwrap_or_default()
-}
-
-pub fn http_client_builder() -> ClientBuilder {
-    Client::builder()
+pub fn create_http_client(timeout: u64, proxy: String) -> Result<Client, String> {
+    let mut client_builder = Client::builder()
         .danger_accept_invalid_certs(true)
         .use_rustls_tls()
+        .timeout(Duration::from_secs(timeout));
+    if !proxy.is_empty() {
+        match Proxy::all(proxy) {
+            Ok(p) => {
+                client_builder = client_builder.proxy(p);
+            }
+            Err(err) => {
+                return Err(err.to_string());
```

Comment:
What was the reason to return a String instead of an Error?

Reply:
That's because the final error messages shown as a result are string. But perhaps there is a more idiomatic way to handle this error. Please send a PR if you find a better way to do it.